FFmpeg.AutoGen/FFmpeg.cs

```diff
@@ -51,9 +51,60 @@ static ffmpeg()
         public static string RootPath { get; set; } = AppDomain.CurrentDomain.BaseDirectory;
 
         public static GetOrLoadLibrary GetOrLoadLibrary { get; set; }
+        
+        /// <summary>
+        ///     Tries to load the native libraries from the set root path. <br/>
+        ///     You can specify which libraries need to be loaded with LibraryFlags.
+        ///     It will try to load all librares by default. <br/>
+        ///     Ideally, you would want to only call this function once, before doing anything with FFmpeg.
+        ///     If you try to do that later, it might unload all of your already loaded libraries and fail to provide them again.
+        /// </summary>
+        /// <returns>Whether it succeeded in loading all the requested libraries.</returns>
+        public static bool TryLoadLibraries(LibraryFlags libraries = LibraryFlags.All, string path = "")
+        {
+            UnloadLibraries();
+            foreach (var libraryName in libraries.ToStrings())
+            {
+                if (LoadLibrary(libraryName, false, path) == IntPtr.Zero)
```

Comment:
This is why it needs to unload libraries. What it is doing when "trying" to load libraries is _actually_ loading them if they exist, and put them in the dictionary of loaded libraries. And when one requested required library is missing, it will return an `IntPtr.Zero`.

Now what if it loaded some of the libraries for the current root path, but not all of them? This is of course a pretty unlikely scenario to happen, since if one installs ffmpeg, all these libraries are supposedly installed - but it still might happen. And in that case, this function, which is intended to be run multiple times, is going to keep libraries loaded ffmpeg previous root paths, which I assume isn't desirable.
So there is a need to unload libraries that may have been loaded.

Luckily though, the most likely scenario is that the dictionary of loaded libraries is already empty and nothing needs to be unloaded.

But maybe there is a way to just check if it can load them, instead of actually loading them? For now though I'm sure this solution will work fine. After all it only needs to be ran once or a couple times on startup.

Reply:
Don't get me wrong - I do appreciate the work you did. However, if you need such elaborative way of probing just override ```GetOrLoadLibrary``` do all this in your project.

Reply:
As I said in my comment above, I wouldn't be the only one in need of this.
Everyone wants to be able to try out different possible paths, after all I'm not the one who forwarded the issue!

Though if you have an idea of how I could check if I can load the library from the current root path without actually doing so, I can surely modify the current code.

Maybe simply checking if the files are there? The problem with that though is that it's platform specific... But when I read some of the code the different names for each OS were there. Maybe the problem isn't that significant.